crates/bevy_math/src/primitives/dim3.rs

```diff
@@ -214,6 +215,80 @@ impl InfinitePlane3d {
 
         (Self { normal }, translation)
     }
+
+    /// Computes the shortest distance between a plane transformed with the given `isometry` and a
+    /// `point`.
+    #[inline]
+    pub fn signed_distance(&self, isometry: Isometry3d, point: Vec3) -> f32 {
+        (isometry * self.normal).dot(isometry * Vec3::ZERO - point)
+    }
+
+    /// Injects the `point` into the plane transformed with the given `isometry`.
+    ///
+    /// This projects the point orthogonally along the shortest path onto the plane.
+    #[inline]
+    pub fn project_point(&self, isometry: Isometry3d, point: Vec3) -> Vec3 {
+        point + self.normal * self.signed_distance(isometry, point)
+    }
+
+    /// Computes the shortest distance between a plane going through `Vec3::ZERO` and a `point`.
+    #[inline]
+    pub fn local_signed_distance(&self, point: Vec3) -> f32 {
+        self.normal.dot(point)
+    }
+
+    /// Injects the `point` into the plane going through `Vec3::ZERO`.
+    ///
+    /// This projects the point orthogonally along the shortest path onto the plane.
+    #[inline]
+    pub fn local_project_point(&self, point: Vec3) -> Vec3 {
+        point + self.normal * self.local_signed_distance(point)
+    }
+
+    /// Computes an [`Affine3A`] which transforms points from the plane in 3D space with the given
+    /// `origin` to the XY-plane.
+    ///
+    /// The transformation is [congruent](https://en.wikipedia.org/wiki/Congruence_(geometry))
+    /// meaning it will preserve shapes. This is **not** the case with normal projections.
+    ///
+    /// See [`InfinitePlane3d::congruent_transformations_xy`] for more information.
+    #[inline]
+    pub fn congruent_projection_xy(&self, origin: Vec3) -> Affine3A {
```

Comment:
I feel that the word projection is misleading here, we're doing a rigid motion (Is this an Isometry? I think so.) to map congruent shapes, not an orthogonal (or a different) projection.

Reply:
I use the following definition: In 3D geometry a projection is

> any method of mapping three-dimensional points to a two-dimensional plane

(Wikipedia)

It doesn't necessarily have to be orthogonal

Reply:
Well, yes, but that's not what the Affine3D does, if you put in a random point in 3D space, you get again, a random point in 3D space, only points inside one specific plane (the one we're calling the method on if I understand correctly) get mapped to the xy plane.

Reply:
Hmm, yeah that's true. So it's just a transformation of the plane. I also struggled to find a good name here and I was pretty happy with the projection <-> injection duality. But yeah. I agree that it's kind of misleading.

Do you have a suggestion?

Reply:
In the end I just called it `isometry_into_xy` and `isometry_from_xy`. I guess there is no real mathematical term for this kind of projection.

Thanks for the review and the discussion!